### PR TITLE
fix: remove tokio_unstable

### PR DESCRIPTION
**Pull request by Kodylow**
Removes tokio_unstable `Builder`, turns the thread name into a tracing span, still enforces using fedimint_core's spawn and spawn_local.

### CODE REVIEW

File: .cargo/config.toml [hunk 1]
```diff
@@ -1,2 +1,2 @@
 [build]
-rustflags = ["--cfg", "tokio_unstable"]
+# rustflags = ["--cfg", "tokio_unstable"]
```
Comment (dpc) [inline .cargo/config.toml:2]: Don't be shy. Just delete it. :D

File: .config/semgrep.yaml [hunk 2]
```diff
@@ -64,6 +64,9 @@ rules:
   message: "`tokio::spawn` won't set task name by default. use `fedimint_core::runtime::spawn` instead"
   pattern: tokio::spawn
   severity: WARNING
+  paths:
```
Comment (dpc) [inline .config/semgrep.yaml:67]: Why?

File: flake.nix [hunk 3]
```diff
@@ -104,7 +104,7 @@
           };
 
           toolchainArgs = {
-            extraRustFlags = "--cfg tokio_unstable";
+            # extraRustFlags = "--cfg tokio_unstable";
```
Comment (dpc) [inline flake.nix:107]: Same here. :)

**Review by dpc — DISMISSED** [review]
Fine with me. Not worth the downstream trouble.

**Review by dpc — COMMENTED** [review]
But maybe clear the nits why we wait for a second opinion.

File: .config/semgrep.yaml (hunk 2 above)
Comment (dpc) [reply .config/semgrep.yaml:67]: Oh.

You can use a `// nosemgrep: <rule-name>` to just exclude specific places that need to break the rule.

Comment (Kodylow) [reply .config/semgrep.yaml]: done

**Review by dpc — APPROVED** [review]

**Review by justinmoon — APPROVED** [review]
LGTM. Seemed like everyone was in favor this morning in the dev call.

**Comment by fedimint-backports** [comment]
Successfully created backport PR for `releases/v0.3`:
- #5213

**Pull request merged**